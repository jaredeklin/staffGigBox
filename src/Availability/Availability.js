import React, { Component } from 'react';
import DayPicker, { DateUtils } from 'react-day-picker';
import 'react-day-picker/lib/style.css';
import { Api } from '../Api/Api'

export class Availability extends Component {
  constructor(props) {
    super(props);
    this.api = new Api();
    this.state = {
      selectedDays: [],
    };
  }

  handleDayClick = (day, { selected }) => {
    const { selectedDays } = this.state;

    if (selected) {
      const selectedIndex = selectedDays.findIndex(selectedDay =>
        DateUtils.isSameDay(selectedDay, day)
      );
      selectedDays.splice(selectedIndex, 1);
    } else {
      selectedDays.push(day);
    }
    this.setState({ selectedDays });
  }

  handleSubmit = () => {

    const dates = this.state.selectedDays.map(day => this.api.cleanDate(day))
    const daysOff = {
      staff_id: this.props.currentUserId,
      days_requested_off: dates
    }
    console.log(daysOff)
  }

  render() {
    return (
      <div>
        <h2>Select days you are unavailable</h2>
        <DayPicker
          selectedDays={ this.state.selectedDays }
          onDayClick={ this.handleDayClick }
        />
        <button onClick={ this.handleSubmit }>Submit Availability</button>
      </div>
    );
  }
}