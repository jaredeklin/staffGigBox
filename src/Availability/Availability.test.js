import React from 'react';
import { shallow } from 'enzyme';
import { Availability } from './Availability';
const moment = require('moment');
const MockDate = require('mockdate');

describe('Availability', () => {
  MockDate.set(moment('2018-06-29T06:00:00.000Z'));
  let wrapper;
  let mockCurrentUserId = 2;
  const mockSelectedDay = 'Fri Jun 30 2018 12:00:00 GMT-0700';

  beforeEach(() => {
    wrapper = shallow(<Availability currentUserId={mockCurrentUserId} />, {
      disableLifecycleMethods: true
    });
    wrapper.setState({ selectedDays: [mockSelectedDay] });
  });

  describe('handleDayClick', () => {
    it('should update state when selected is undefined', () => {
      const mockSelected = { selected: undefined };
      const expected = [mockSelectedDay, mockSelectedDay];
      wrapper.instance().handleDayClick(mockSelectedDay, mockSelected);

      expect(wrapper.state().selectedDays).toEqual(expected);
    });

    it('should update state when selected is true', async () => {
      const mockSelected = { selected: true };
      wrapper.setState({ selectedDays: [new Date()] });
      wrapper
        .instance()
        .handleDayClick(new Date(mockSelectedDay), mockSelected);

      expect(wrapper.state('selectedDays')).toEqual([]);
    });
  });

  describe('handleSubmit', () => {
    it('should call getAvailability with correct params', () => {
      wrapper.instance().api.getAvailability = jest.fn().mockReturnValue(false);
      wrapper.instance().api.postAvailability = jest.fn();
      wrapper.instance().cleanDate = jest.fn(() => 'Jun 30, 2018');

      wrapper.instance().handleSubmit();

      expect(wrapper.instance().cleanDate).toHaveBeenCalledWith(
        mockSelectedDay
      );
      expect(wrapper.instance().api.getAvailability).toHaveBeenCalledWith(
        2,
        'Jun 30, 2018'
      );
    });
  });

  describe('cleanDate', () => {
    it('should return a parsed date', () => {
      const mockDay = 'Fri Jun 30 2018 12:00:00 GMT-0700';
      const expected = 'Jun 30, 2018';

      expect(wrapper.instance().cleanDate(mockDay)).toEqual(expected);
    });
  });

  describe('componentDidMount', () => {
    const mockDate = [{ date_unavailable: 'June 30, 2018' }];

    it('should call getAvailability', () => {
      wrapper.instance().api.getAvailability = jest
        .fn()
        .mockReturnValue(mockDate);
      wrapper.instance().componentDidMount();

      expect(wrapper.instance().api.getAvailability).toHaveBeenCalledWith(
        mockCurrentUserId
      );
    });

    it('should setState with selectedDays', () => {
      wrapper.instance().api.getAvailability = jest
        .fn()
        .mockReturnValue(mockDate);
      wrapper.instance().componentDidMount();

      expect(wrapper.state('selectedDays')).toEqual([mockSelectedDay]);
    });
  });

  describe('handleDayClick', () => {
    it('should setState for selectedDays', () => {
      // wrapper.instance().handleDayClick()
      expect(wrapper.state()).toEqual({
        selectedDays: [mockSelectedDay]
      });
    });
  });
});
