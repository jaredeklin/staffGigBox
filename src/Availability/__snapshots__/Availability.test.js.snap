// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Availability should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Availability />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Select days you are unavailable
</h2>,
        <DayPicker
          canChangeMonth={true}
          captionElement={
                    <Caption
                              classNames={
                                        Object {
                                                  "body": "DayPicker-Body",
                                                  "caption": "DayPicker-Caption",
                                                  "container": "DayPicker",
                                                  "day": "DayPicker-Day",
                                                  "disabled": "disabled",
                                                  "footer": "DayPicker-Footer",
                                                  "interactionDisabled": "DayPicker--interactionDisabled",
                                                  "month": "DayPicker-Month",
                                                  "months": "DayPicker-Months",
                                                  "navBar": "DayPicker-NavBar",
                                                  "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                                  "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                                  "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                                  "outside": "outside",
                                                  "selected": "selected",
                                                  "today": "today",
                                                  "todayButton": "DayPicker-TodayButton",
                                                  "week": "DayPicker-Week",
                                                  "weekNumber": "DayPicker-WeekNumber",
                                                  "weekday": "DayPicker-Weekday",
                                                  "weekdays": "DayPicker-Weekdays",
                                                  "weekdaysRow": "DayPicker-WeekdaysRow",
                                                  "wrapper": "DayPicker-wrapper",
                                                }
                              }
                              localeUtils={
                                        Object {
                                                  "formatDay": [Function],
                                                  "formatMonthTitle": [Function],
                                                  "formatWeekdayLong": [Function],
                                                  "formatWeekdayShort": [Function],
                                                  "getFirstDayOfWeek": [Function],
                                                  "getMonths": [Function],
                                                }
                              }
                    />
          }
          classNames={
                    Object {
                              "body": "DayPicker-Body",
                              "caption": "DayPicker-Caption",
                              "container": "DayPicker",
                              "day": "DayPicker-Day",
                              "disabled": "disabled",
                              "footer": "DayPicker-Footer",
                              "interactionDisabled": "DayPicker--interactionDisabled",
                              "month": "DayPicker-Month",
                              "months": "DayPicker-Months",
                              "navBar": "DayPicker-NavBar",
                              "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                              "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                              "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                              "outside": "outside",
                              "selected": "selected",
                              "today": "today",
                              "todayButton": "DayPicker-TodayButton",
                              "week": "DayPicker-Week",
                              "weekNumber": "DayPicker-WeekNumber",
                              "weekday": "DayPicker-Weekday",
                              "weekdays": "DayPicker-Weekdays",
                              "weekdaysRow": "DayPicker-WeekdaysRow",
                              "wrapper": "DayPicker-wrapper",
                            }
          }
          enableOutsideDaysClick={true}
          fixedWeeks={false}
          initialMonth={2018-06-18T17:24:26.412Z}
          labels={
                    Object {
                              "nextMonth": "Next Month",
                              "previousMonth": "Previous Month",
                            }
          }
          locale="en"
          localeUtils={
                    Object {
                              "default": Object {
                                "formatDay": [Function],
                                "formatMonthTitle": [Function],
                                "formatWeekdayLong": [Function],
                                "formatWeekdayShort": [Function],
                                "getFirstDayOfWeek": [Function],
                                "getMonths": [Function],
                              },
                              "formatDay": [Function],
                              "formatMonthTitle": [Function],
                              "formatWeekdayLong": [Function],
                              "formatWeekdayShort": [Function],
                              "getFirstDayOfWeek": [Function],
                              "getMonths": [Function],
                            }
          }
          navbarElement={
                    <Navbar
                              classNames={
                                        Object {
                                                  "body": "DayPicker-Body",
                                                  "caption": "DayPicker-Caption",
                                                  "container": "DayPicker",
                                                  "day": "DayPicker-Day",
                                                  "disabled": "disabled",
                                                  "footer": "DayPicker-Footer",
                                                  "interactionDisabled": "DayPicker--interactionDisabled",
                                                  "month": "DayPicker-Month",
                                                  "months": "DayPicker-Months",
                                                  "navBar": "DayPicker-NavBar",
                                                  "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                                  "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                                  "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                                  "outside": "outside",
                                                  "selected": "selected",
                                                  "today": "today",
                                                  "todayButton": "DayPicker-TodayButton",
                                                  "week": "DayPicker-Week",
                                                  "weekNumber": "DayPicker-WeekNumber",
                                                  "weekday": "DayPicker-Weekday",
                                                  "weekdays": "DayPicker-Weekdays",
                                                  "weekdaysRow": "DayPicker-WeekdaysRow",
                                                  "wrapper": "DayPicker-wrapper",
                                                }
                              }
                              dir="ltr"
                              labels={
                                        Object {
                                                  "nextMonth": "Next Month",
                                                  "previousMonth": "Previous Month",
                                                }
                              }
                              showNextButton={true}
                              showPreviousButton={true}
                    />
          }
          numberOfMonths={1}
          onDayClick={[Function]}
          pagedNavigation={false}
          renderDay={[Function]}
          renderWeek={[Function]}
          reverseMonths={false}
          selectedDays={Array []}
          showOutsideDays={false}
          showWeekDays={true}
          showWeekNumbers={false}
          tabIndex={0}
          weekdayElement={<Weekday />}
/>,
        <button
          onClick={[Function]}
>
          Submit Availability
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select days you are unavailable",
        },
        "ref": null,
        "rendered": "Select days you are unavailable",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "canChangeMonth": true,
          "captionElement": <Caption
            classNames={
                        Object {
                                    "body": "DayPicker-Body",
                                    "caption": "DayPicker-Caption",
                                    "container": "DayPicker",
                                    "day": "DayPicker-Day",
                                    "disabled": "disabled",
                                    "footer": "DayPicker-Footer",
                                    "interactionDisabled": "DayPicker--interactionDisabled",
                                    "month": "DayPicker-Month",
                                    "months": "DayPicker-Months",
                                    "navBar": "DayPicker-NavBar",
                                    "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                    "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                    "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                    "outside": "outside",
                                    "selected": "selected",
                                    "today": "today",
                                    "todayButton": "DayPicker-TodayButton",
                                    "week": "DayPicker-Week",
                                    "weekNumber": "DayPicker-WeekNumber",
                                    "weekday": "DayPicker-Weekday",
                                    "weekdays": "DayPicker-Weekdays",
                                    "weekdaysRow": "DayPicker-WeekdaysRow",
                                    "wrapper": "DayPicker-wrapper",
                                  }
            }
            localeUtils={
                        Object {
                                    "formatDay": [Function],
                                    "formatMonthTitle": [Function],
                                    "formatWeekdayLong": [Function],
                                    "formatWeekdayShort": [Function],
                                    "getFirstDayOfWeek": [Function],
                                    "getMonths": [Function],
                                  }
            }
/>,
          "classNames": Object {
            "body": "DayPicker-Body",
            "caption": "DayPicker-Caption",
            "container": "DayPicker",
            "day": "DayPicker-Day",
            "disabled": "disabled",
            "footer": "DayPicker-Footer",
            "interactionDisabled": "DayPicker--interactionDisabled",
            "month": "DayPicker-Month",
            "months": "DayPicker-Months",
            "navBar": "DayPicker-NavBar",
            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
            "outside": "outside",
            "selected": "selected",
            "today": "today",
            "todayButton": "DayPicker-TodayButton",
            "week": "DayPicker-Week",
            "weekNumber": "DayPicker-WeekNumber",
            "weekday": "DayPicker-Weekday",
            "weekdays": "DayPicker-Weekdays",
            "weekdaysRow": "DayPicker-WeekdaysRow",
            "wrapper": "DayPicker-wrapper",
          },
          "enableOutsideDaysClick": true,
          "fixedWeeks": false,
          "initialMonth": 2018-06-18T17:24:26.412Z,
          "labels": Object {
            "nextMonth": "Next Month",
            "previousMonth": "Previous Month",
          },
          "locale": "en",
          "localeUtils": Object {
            "default": Object {
              "formatDay": [Function],
              "formatMonthTitle": [Function],
              "formatWeekdayLong": [Function],
              "formatWeekdayShort": [Function],
              "getFirstDayOfWeek": [Function],
              "getMonths": [Function],
            },
            "formatDay": [Function],
            "formatMonthTitle": [Function],
            "formatWeekdayLong": [Function],
            "formatWeekdayShort": [Function],
            "getFirstDayOfWeek": [Function],
            "getMonths": [Function],
          },
          "navbarElement": <Navbar
            classNames={
                        Object {
                                    "body": "DayPicker-Body",
                                    "caption": "DayPicker-Caption",
                                    "container": "DayPicker",
                                    "day": "DayPicker-Day",
                                    "disabled": "disabled",
                                    "footer": "DayPicker-Footer",
                                    "interactionDisabled": "DayPicker--interactionDisabled",
                                    "month": "DayPicker-Month",
                                    "months": "DayPicker-Months",
                                    "navBar": "DayPicker-NavBar",
                                    "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                    "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                    "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                    "outside": "outside",
                                    "selected": "selected",
                                    "today": "today",
                                    "todayButton": "DayPicker-TodayButton",
                                    "week": "DayPicker-Week",
                                    "weekNumber": "DayPicker-WeekNumber",
                                    "weekday": "DayPicker-Weekday",
                                    "weekdays": "DayPicker-Weekdays",
                                    "weekdaysRow": "DayPicker-WeekdaysRow",
                                    "wrapper": "DayPicker-wrapper",
                                  }
            }
            dir="ltr"
            labels={
                        Object {
                                    "nextMonth": "Next Month",
                                    "previousMonth": "Previous Month",
                                  }
            }
            showNextButton={true}
            showPreviousButton={true}
/>,
          "numberOfMonths": 1,
          "onDayClick": [Function],
          "pagedNavigation": false,
          "renderDay": [Function],
          "renderWeek": [Function],
          "reverseMonths": false,
          "selectedDays": Array [],
          "showOutsideDays": false,
          "showWeekDays": true,
          "showWeekNumbers": false,
          "tabIndex": 0,
          "weekdayElement": <Weekday />,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit Availability",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit Availability",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Select days you are unavailable
</h2>,
          <DayPicker
            canChangeMonth={true}
            captionElement={
                        <Caption
                                    classNames={
                                                Object {
                                                            "body": "DayPicker-Body",
                                                            "caption": "DayPicker-Caption",
                                                            "container": "DayPicker",
                                                            "day": "DayPicker-Day",
                                                            "disabled": "disabled",
                                                            "footer": "DayPicker-Footer",
                                                            "interactionDisabled": "DayPicker--interactionDisabled",
                                                            "month": "DayPicker-Month",
                                                            "months": "DayPicker-Months",
                                                            "navBar": "DayPicker-NavBar",
                                                            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                                            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                                            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                                            "outside": "outside",
                                                            "selected": "selected",
                                                            "today": "today",
                                                            "todayButton": "DayPicker-TodayButton",
                                                            "week": "DayPicker-Week",
                                                            "weekNumber": "DayPicker-WeekNumber",
                                                            "weekday": "DayPicker-Weekday",
                                                            "weekdays": "DayPicker-Weekdays",
                                                            "weekdaysRow": "DayPicker-WeekdaysRow",
                                                            "wrapper": "DayPicker-wrapper",
                                                          }
                                    }
                                    localeUtils={
                                                Object {
                                                            "formatDay": [Function],
                                                            "formatMonthTitle": [Function],
                                                            "formatWeekdayLong": [Function],
                                                            "formatWeekdayShort": [Function],
                                                            "getFirstDayOfWeek": [Function],
                                                            "getMonths": [Function],
                                                          }
                                    }
                        />
            }
            classNames={
                        Object {
                                    "body": "DayPicker-Body",
                                    "caption": "DayPicker-Caption",
                                    "container": "DayPicker",
                                    "day": "DayPicker-Day",
                                    "disabled": "disabled",
                                    "footer": "DayPicker-Footer",
                                    "interactionDisabled": "DayPicker--interactionDisabled",
                                    "month": "DayPicker-Month",
                                    "months": "DayPicker-Months",
                                    "navBar": "DayPicker-NavBar",
                                    "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                    "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                    "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                    "outside": "outside",
                                    "selected": "selected",
                                    "today": "today",
                                    "todayButton": "DayPicker-TodayButton",
                                    "week": "DayPicker-Week",
                                    "weekNumber": "DayPicker-WeekNumber",
                                    "weekday": "DayPicker-Weekday",
                                    "weekdays": "DayPicker-Weekdays",
                                    "weekdaysRow": "DayPicker-WeekdaysRow",
                                    "wrapper": "DayPicker-wrapper",
                                  }
            }
            enableOutsideDaysClick={true}
            fixedWeeks={false}
            initialMonth={2018-06-18T17:24:26.412Z}
            labels={
                        Object {
                                    "nextMonth": "Next Month",
                                    "previousMonth": "Previous Month",
                                  }
            }
            locale="en"
            localeUtils={
                        Object {
                                    "default": Object {
                                      "formatDay": [Function],
                                      "formatMonthTitle": [Function],
                                      "formatWeekdayLong": [Function],
                                      "formatWeekdayShort": [Function],
                                      "getFirstDayOfWeek": [Function],
                                      "getMonths": [Function],
                                    },
                                    "formatDay": [Function],
                                    "formatMonthTitle": [Function],
                                    "formatWeekdayLong": [Function],
                                    "formatWeekdayShort": [Function],
                                    "getFirstDayOfWeek": [Function],
                                    "getMonths": [Function],
                                  }
            }
            navbarElement={
                        <Navbar
                                    classNames={
                                                Object {
                                                            "body": "DayPicker-Body",
                                                            "caption": "DayPicker-Caption",
                                                            "container": "DayPicker",
                                                            "day": "DayPicker-Day",
                                                            "disabled": "disabled",
                                                            "footer": "DayPicker-Footer",
                                                            "interactionDisabled": "DayPicker--interactionDisabled",
                                                            "month": "DayPicker-Month",
                                                            "months": "DayPicker-Months",
                                                            "navBar": "DayPicker-NavBar",
                                                            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                                            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                                            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                                            "outside": "outside",
                                                            "selected": "selected",
                                                            "today": "today",
                                                            "todayButton": "DayPicker-TodayButton",
                                                            "week": "DayPicker-Week",
                                                            "weekNumber": "DayPicker-WeekNumber",
                                                            "weekday": "DayPicker-Weekday",
                                                            "weekdays": "DayPicker-Weekdays",
                                                            "weekdaysRow": "DayPicker-WeekdaysRow",
                                                            "wrapper": "DayPicker-wrapper",
                                                          }
                                    }
                                    dir="ltr"
                                    labels={
                                                Object {
                                                            "nextMonth": "Next Month",
                                                            "previousMonth": "Previous Month",
                                                          }
                                    }
                                    showNextButton={true}
                                    showPreviousButton={true}
                        />
            }
            numberOfMonths={1}
            onDayClick={[Function]}
            pagedNavigation={false}
            renderDay={[Function]}
            renderWeek={[Function]}
            reverseMonths={false}
            selectedDays={Array []}
            showOutsideDays={false}
            showWeekDays={true}
            showWeekNumbers={false}
            tabIndex={0}
            weekdayElement={<Weekday />}
/>,
          <button
            onClick={[Function]}
>
            Submit Availability
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select days you are unavailable",
          },
          "ref": null,
          "rendered": "Select days you are unavailable",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "canChangeMonth": true,
            "captionElement": <Caption
              classNames={
                            Object {
                                          "body": "DayPicker-Body",
                                          "caption": "DayPicker-Caption",
                                          "container": "DayPicker",
                                          "day": "DayPicker-Day",
                                          "disabled": "disabled",
                                          "footer": "DayPicker-Footer",
                                          "interactionDisabled": "DayPicker--interactionDisabled",
                                          "month": "DayPicker-Month",
                                          "months": "DayPicker-Months",
                                          "navBar": "DayPicker-NavBar",
                                          "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                          "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                          "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                          "outside": "outside",
                                          "selected": "selected",
                                          "today": "today",
                                          "todayButton": "DayPicker-TodayButton",
                                          "week": "DayPicker-Week",
                                          "weekNumber": "DayPicker-WeekNumber",
                                          "weekday": "DayPicker-Weekday",
                                          "weekdays": "DayPicker-Weekdays",
                                          "weekdaysRow": "DayPicker-WeekdaysRow",
                                          "wrapper": "DayPicker-wrapper",
                                        }
              }
              localeUtils={
                            Object {
                                          "formatDay": [Function],
                                          "formatMonthTitle": [Function],
                                          "formatWeekdayLong": [Function],
                                          "formatWeekdayShort": [Function],
                                          "getFirstDayOfWeek": [Function],
                                          "getMonths": [Function],
                                        }
              }
/>,
            "classNames": Object {
              "body": "DayPicker-Body",
              "caption": "DayPicker-Caption",
              "container": "DayPicker",
              "day": "DayPicker-Day",
              "disabled": "disabled",
              "footer": "DayPicker-Footer",
              "interactionDisabled": "DayPicker--interactionDisabled",
              "month": "DayPicker-Month",
              "months": "DayPicker-Months",
              "navBar": "DayPicker-NavBar",
              "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
              "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
              "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
              "outside": "outside",
              "selected": "selected",
              "today": "today",
              "todayButton": "DayPicker-TodayButton",
              "week": "DayPicker-Week",
              "weekNumber": "DayPicker-WeekNumber",
              "weekday": "DayPicker-Weekday",
              "weekdays": "DayPicker-Weekdays",
              "weekdaysRow": "DayPicker-WeekdaysRow",
              "wrapper": "DayPicker-wrapper",
            },
            "enableOutsideDaysClick": true,
            "fixedWeeks": false,
            "initialMonth": 2018-06-18T17:24:26.412Z,
            "labels": Object {
              "nextMonth": "Next Month",
              "previousMonth": "Previous Month",
            },
            "locale": "en",
            "localeUtils": Object {
              "default": Object {
                "formatDay": [Function],
                "formatMonthTitle": [Function],
                "formatWeekdayLong": [Function],
                "formatWeekdayShort": [Function],
                "getFirstDayOfWeek": [Function],
                "getMonths": [Function],
              },
              "formatDay": [Function],
              "formatMonthTitle": [Function],
              "formatWeekdayLong": [Function],
              "formatWeekdayShort": [Function],
              "getFirstDayOfWeek": [Function],
              "getMonths": [Function],
            },
            "navbarElement": <Navbar
              classNames={
                            Object {
                                          "body": "DayPicker-Body",
                                          "caption": "DayPicker-Caption",
                                          "container": "DayPicker",
                                          "day": "DayPicker-Day",
                                          "disabled": "disabled",
                                          "footer": "DayPicker-Footer",
                                          "interactionDisabled": "DayPicker--interactionDisabled",
                                          "month": "DayPicker-Month",
                                          "months": "DayPicker-Months",
                                          "navBar": "DayPicker-NavBar",
                                          "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                                          "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                                          "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                                          "outside": "outside",
                                          "selected": "selected",
                                          "today": "today",
                                          "todayButton": "DayPicker-TodayButton",
                                          "week": "DayPicker-Week",
                                          "weekNumber": "DayPicker-WeekNumber",
                                          "weekday": "DayPicker-Weekday",
                                          "weekdays": "DayPicker-Weekdays",
                                          "weekdaysRow": "DayPicker-WeekdaysRow",
                                          "wrapper": "DayPicker-wrapper",
                                        }
              }
              dir="ltr"
              labels={
                            Object {
                                          "nextMonth": "Next Month",
                                          "previousMonth": "Previous Month",
                                        }
              }
              showNextButton={true}
              showPreviousButton={true}
/>,
            "numberOfMonths": 1,
            "onDayClick": [Function],
            "pagedNavigation": false,
            "renderDay": [Function],
            "renderWeek": [Function],
            "reverseMonths": false,
            "selectedDays": Array [],
            "showOutsideDays": false,
            "showWeekDays": true,
            "showWeekNumbers": false,
            "tabIndex": 0,
            "weekdayElement": <Weekday />,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit Availability",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit Availability",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
